From: "roberto@quantal64" <roberto@quantal64>
Date: Tue Aug 14 08:15:43 2012 +0200
Subject: applied Aarni Koskela patch for mod_uwsgi
Description: Fix mod_uwsgi authenticated, but authentication typeless requests
 In some configurations - case in point: OpenAM web agent
 authentication - an Apache request may contain an user (r->user) but no valid
 authentication type. This eventually leads into a strlen(NULL), which will
 segfault the Apache worker.
Origin: http://projects.unbit.it/hg/uwsgi-1.2/rev/8ee4797038aa

diff -r b370adcb3544 -r a7366f4fd63b apache2/mod_uwsgi.c
--- a/apache2/mod_uwsgi.c	Tue Aug 14 08:14:16 2012 +0200
+++ b/apache2/mod_uwsgi.c	Tue Aug 14 08:15:43 2012 +0200
@@ -367,7 +367,7 @@
 		vecptr = uwsgi_add_var(uwsgi_vars, vecptr, r, "REMOTE_USER", "", &pkt_size) ;
 	}
 
-	if (r->user) {
+	if (ap_auth_type(r) != NULL) {
 		vecptr = uwsgi_add_var(uwsgi_vars, vecptr, r, "AUTH_TYPE", (char *) ap_auth_type(r), &pkt_size) ;
 	}
 	vecptr = uwsgi_add_var(uwsgi_vars, vecptr, r, "DOCUMENT_ROOT", (char *) ap_document_root(r), &pkt_size) ;
